pkg/openstack/loadbalancer_test.go

```diff
@@ -1629,3 +1632,277 @@ func TestLbaasV2_getMemberSubnetID(t *testing.T) {
+						},
+					},
+				},
+				service: &corev1.Service{},
+			},
+			want: "test-default-subnet-id",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := tt.args.lbaasV2.getSubnetID(tt.args.service, tt.args.svcConf)
+			if tt.expectedErr != "" {
+				assert.EqualError(t, err, tt.expectedErr)
+			}
+			if tt.expectedErr == "" {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, got, tt.want)
+		})
+	}
+}
+
+func Test_buildMonitorCreateOpts(t *testing.T) {
+	type testArg struct {
+		lbaas   *LbaasV2
+		svcConf *serviceConfig
+		port    corev1.ServicePort
+	}
+	tests := []struct {
+		name    string
+		testArg testArg
+		want    v2monitors.CreateOpts
+	}{
+		{
+			name: "test for port protocol udp with ovn provider",
+			testArg: testArg{
+				lbaas: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBProvider: "ovn",
+						},
+						lb: &gophercloud.ServiceClient{},
+					},
+				},
+				svcConf: &serviceConfig{
+					healthMonitorDelay:          6,
+					healthMonitorTimeout:        5,
+					healthMonitorMaxRetries:     4,
+					healthMonitorMaxRetriesDown: 3,
+					healthCheckNodePort:         32100,
+				},
+				port: corev1.ServicePort{
+					Protocol: corev1.ProtocolUDP,
+				},
+			},
+			want: v2monitors.CreateOpts{
+				Name:           "test for port protocol udp with ovn provider",
+				Type:           "UDP-CONNECT",
+				Delay:          6,
+				Timeout:        5,
+				MaxRetries:     4,
+				MaxRetriesDown: 3,
+			},
+		},
+		{
+			name: "using tcp with ovn provider",
+			testArg: testArg{
+				lbaas: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBProvider: "ovn",
+						},
+					},
+				},
+				svcConf: &serviceConfig{
+					healthMonitorDelay:          3,
+					healthMonitorTimeout:        8,
+					healthMonitorMaxRetries:     6,
+					healthMonitorMaxRetriesDown: 2,
+					healthCheckNodePort:         31200,
+				},
+				port: corev1.ServicePort{
+					Protocol: corev1.ProtocolTCP,
+				},
+			},
+			want: v2monitors.CreateOpts{
+				Name:           "using tcp with ovn provider",
+				Type:           "TCP",
+				Delay:          3,
+				Timeout:        8,
+				MaxRetries:     6,
+				MaxRetriesDown: 2,
+			},
+		},
+		{
+			name: "using udp with ovn provider",
+			testArg: testArg{
+				lbaas: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBProvider: "ovn",
+						},
+					},
+				},
+				svcConf: &serviceConfig{
+					healthMonitorDelay:          8,
+					healthMonitorTimeout:        4,
+					healthMonitorMaxRetries:     3,
+					healthMonitorMaxRetriesDown: 2,
+					healthCheckNodePort:         32000,
+				},
+				port: corev1.ServicePort{
+					Protocol: corev1.ProtocolUDP,
+				},
+			},
+			want: v2monitors.CreateOpts{
+				Name:           "using udp with ovn provider",
+				Type:           "UDP-CONNECT",
+				Delay:          8,
+				Timeout:        4,
+				MaxRetries:     3,
+				MaxRetriesDown: 2,
+			},
+		},
```

Comment:
How is this different from the first test case?

Reply:
True! nice catch, i was thinking of way to test the gophercloud part of the function, but it was tricky to mock so I go removed it and forgot both test cases were thesame without that